pkg/controller/client.go

```diff
@@ -173,5 +174,16 @@ func (c *XdsClient) closeStreamClient() {
 }
 
 func (c *XdsClient) Close() error {
+	if restart.GetExitType() == restart.Restart {
+		hashName := &ads.HashName{
+			NameToCds: *c.AdsController.Processor.Cache.ClusterCache.GetClusterHashPtr(),
+			NameToLds: *c.AdsController.Processor.Cache.ListenerCache.GetListenerHashPtr(),
+			NameToRds: *c.AdsController.Processor.Cache.RouteCache.GetRouteHashPtr(),
+		}
+		ads.ResetPersistFile()
```

Comment:
Where is the hash persistence code in dual-engine mode?

Reply:
https://github.com/kmesh-net/kmesh/blob/288f6faccf38645c17c216eadb52343468844ce1/pkg/controller/workload/workload_processor.go#L808